File: backend/src/main/java/github/lms/lemuel/xr/common/AppException.java
package github.lms.lemuel.xr.common;

import lombok.Getter;

/**
 * Domain·application 계층에서 던지는 비즈니스 예외.
 * ProblemDetailMapper 가 RFC 7807 응답으로 변환.
 */
@Getter
public class AppException extends RuntimeException {

    private final ErrorCode code;

    public AppException(ErrorCode code) {
        super(code.defaultTitle());
        this.code = code;
    }

    public AppException(ErrorCode code, String detail) {
        super(detail);
        this.code = code;
    }

    public AppException(ErrorCode code, String detail, Throwable cause) {
        super(detail, cause);
        this.code = code;
    }
}

File: backend/src/main/kotlin/github/lms/lemuel/xr/common/AppException.kt
package github.lms.lemuel.xr.common

/**
 * Domain·application 계층에서 던지는 비즈니스 예외.
 * ProblemDetailMapper 가 RFC 7807 응답으로 변환.
 */
class AppException : RuntimeException {

    val code: ErrorCode

    constructor(code: ErrorCode) : super(code.defaultTitle) {
        this.code = code
    }

    constructor(code: ErrorCode, detail: String?) : super(detail) {
        this.code = code
    }

    constructor(code: ErrorCode, detail: String?, cause: Throwable?) : super(detail, cause) {
        this.code = code
    }
}
